### ISSUE TITLE
REST API: Can't differentiate a product variation's tax class when it's set to the same as its parent

### ISSUE THREAD

**Issue by jaclync** [woocommerce/woocommerce#27131]
**Prerequisites (mark completed items with an [x]):**
- [x] I have have carried out troubleshooting steps and I believe I have found a bug. (tested on 2 sites)
- [x] I have searched for similar bugs in both open and closed issues and cannot find a duplicate.

**Describe the bug**

When a product variation's tax class is set to "Same as parent" in core, the API field `tax_class` returns the current tax class (either the variation's tax class or its parent product's tax class) and the client cannot tell whether it's set to the same as its parent product.

**Expected behavior**

A different value like `parent` in the `tax_class` field (any value that a slug cannot be set to), an empty value wouldn't work because a product's tax class could be empty when it's set to "Standard". Another field in the API for a variation like `tax_class_same_as_parent: Bool` could also be an option.

**Actual behavior**

`tax_class` in the API returns the same value whether the tax class for a variation is set to "Same as parent" or the same value as its parent product in core.

**Steps to reproduce the bug (We need to be able to reproduce the bug in order to fix it.)**
Steps to reproduce the bug:
1. In wp-admin, go to a variable product with at least one variation
2. In the variation form, set the tax class to "Same as parent" --> in the API response for the variation (`{{site_url}}/wp-json/wc/v3/products/{{product_id}}/variations/{{product_variation_id}}`), the `tax_class` field returns the same value as the `tax_class` field of the parent product (`{{site_url}}/wp-json/wc/v3/products/{{product_id}}`)
3. In the variation form, set the tax class to the same tax class as its parent product --> in the API response for the variation (`{{site_url}}/wp-json/wc/v3/products/{{product_id}}/variations/{{product_variation_id}}`), the `tax_class` field returns the same value as the `tax_class` field of the parent product just like in the previous step

**Isolating the problem (mark completed items with an [x]):**
- [ ] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [ ] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/storefront/).
- [x] I can reproduce this bug consistently using the steps above.

**Comment by joshbetz** [comment]
@jaclync To make sure I understand the problem here, is it that you can't tell if the tax class is inherited from the parent? And the problem is that you then can't edit the tax class of a product that inherits from a parent?

**Comment by jaclync** [comment]
> To make sure I understand the problem here, is it that you can't tell if the tax class is inherited from the parent? And the problem is that you then can't edit the tax class of a product that inherits from a parent?

exactly! we currently don't show the tax/shipping class of a variation in the app because of this. example screenshot in core:

<img src="https://user-images.githubusercontent.com/1945542/134095620-a59c47ce-1ea4-44ac-a3ee-3ed9b5a463a1.png" width="300" />

**Comment by rrennick** [comment]
@jaclync Can you confirm you are still seeing this issue? I tested with

- Parent product with Test tax class
<img width="779" alt="Screen Shot 2021-10-18 at 11 25 33 AM" src="https://user-images.githubusercontent.com/343847/137750544-ad2cdc2c-6802-46e9-bec0-a3c54c7a9601.png">

- Variation set to Same as parent
<img width="852" alt="Screen Shot 2021-10-18 at 11 21 53 AM" src="https://user-images.githubusercontent.com/343847/137750805-420dd173-82a7-4468-a524-2cf1c2328d9e.png">

- Parent product via API - https://ron.localhost/wp-json/wc/v3/products/18705
```
{
  "id": 18705,
  "name": "Gorgeous Linen Bench",
  "slug": "gorgeous-linen-bench",
  "permalink": "https://ron.localhost/product/gorgeous-linen-bench/",
  "date_created": "2021-10-01T14:29:18",
  "date_created_gmt": "2021-10-01T17:29:18",
  "date_modified": "2021-10-18T11:27:01",
  "date_modified_gmt": "2021-10-18T14:27:01",
  "type": "variable",
  "status": "publish",
  "featured": false,
  "catalog_visibility": "visible",
  "description": "",
  "short_description": "",
  "sku": "gorgeous-linen-bench-64818887",
  "price": "48.85",
  "regular_price": "",
  "sale_price": "",
  "date_on_sale_from": null,
  "date_on_sale_from_gmt": null,
  "date_on_sale_to": null,
  "date_on_sale_to_gmt": null,
  "on_sale": false,
  "purchasable": true,
  "total_sales": 0,
  "virtual": false,
  "downloadable": false,
  "downloads": [],
  "download_limit": -1,
  "download_expiry": -1,
  "external_url": "",
  "button_text": "",
  "tax_status": "taxable",
  "tax_class": "test-tax-class",
```

- Variation via API - https://ron.localhost/wp-json/wc/v3/products/18705/variations/18706
```
{
  "id": 18706,
  "date_created": "2021-10-01T14:29:18",
  "date_created_gmt": "2021-10-01T17:29:18",
  "date_modified": "2021-10-18T11:27:01",
  "date_modified_gmt": "2021-10-18T14:27:01",
  "description": "",
  "permalink": "https://ron.localhost/product/gorgeous-linen-bench/?attribute_delectus=sint&attribute_ex=totam",
  "sku": "gorgeous-linen-bench-64818887",
  "price": "612.42",
  "regular_price": "612.42",
  "sale_price": "",
  "date_on_sale_from": null,
  "date_on_sale_from_gmt": null,
  "date_on_sale_to": "2019-04-01T23:59:59",
  "date_on_sale_to_gmt": "2019-04-02T02:59:59",
  "on_sale": false,
  "status": "publish",
  "purchasable": true,
  "virtual": false,
  "downloadable": false,
  "downloads": [],
  "download_limit": -1,
  "download_expiry": -1,
  "tax_status": "taxable",
  "tax_class": "",
```

**Comment by rrennick** [comment]
> @jaclync Can you confirm you are still seeing this issue?

Sorry, please ignore that. I hadn't set the variation tax class to Same as parent.

**Comment by barryhughes** [comment]
An additional field that indicates if the class is same-of-parent, etc, seems like something we can add without any real backward compatibility concerns. Thanks for submitting it!

**Comment by barryhughes** [comment]
@jaclync I've initially set the priority to `low`. Please let me know if you feel this should be elevated (I'm not sure how much of an active concern this is, in relation to your project).

**Comment by jaclync** [comment]
Hi @barryhughes! thanks for taking a look, an additional field that indicates if the class is same-of-parent sounds like a good solution 👍 We haven't heard user reports on this issue, keeping it a low priority is fine 🙂

**Comment by samrat-biswas-git** [comment]
Hi @jaclync - thanks for raising this.

I know this has been open for a very long time, and I appreciate your patience. We're currently going through some older WooCommerce issues, identifying what still needs fixing, and tidying up what's already fixed or out of scope.

This one's really more of an internal process or repo task than a product bug, so I've routed it over to the relevant engineering team to pick up.

Suggested next step: Route to product/API review to choose the public response contract for inherited variation classes, then hand the agreed schema change to engineering.
